README.md

```diff
@@ -7,20 +8,43 @@ This GitHub Action (created using typescript) uses [CLI for Microsoft 365](https
+
 Create a workflow `.yml` file in your `.github/workflows` directory. An [example workflow](#example-workflow---cli-for-microsoft-365-login) is available below. For more information, reference the GitHub Help Documentation for [Creating a workflow file](https://help.github.com/en/articles/configuring-a-workflow#creating-a-workflow-file).
 
 ### Inputs
-- `ADMIN_USERNAME` : **Required** Username (email address of the admin)
-- `ADMIN_PASSWORD` : **Required** Password of the admin
+
+- `ADMIN_USERNAME` : Username (email address of the admin)
+- `ADMIN_PASSWORD` : Password of the admin
+- `CERTIFICATE_ENCODED` : Base64-encoded string with certificate private key
+- `CERTIFICATE_PASSWORD` : Password for the certificate
+- `AAD_APP_ID` : App ID of the Azure AD application to use for certificate authentication
+- `TENANT_ID` : ID or domain (for example "contoso.onmicrosoft.com") of the tenant from which accounts should be able to authenticate
+
+All inputs are optional. But depending of the authentication type chosen, following pair of inputs will be necessary:
+
+- `authType` is `password`: `ADMIN_USERNAME` and `ADMIN_PASSWORD` are required
+- `authType` is `certificate`: at least `CERTIFICATE_ENCODED` and `AAD_APP_ID` are required
+  - `TENANT_ID` will be required. It can be the ID of the tenant from which accounts should be able to authenticate. Use `common` or `organization` if the app is multitenant
+  - Depending on the certificate provided, if encoded with password, `CERTIFICATE_PASSWORD` will be required
 
 #### Optional requirement
-Since this action requires user name and password which are sensitive pieces of information, it would be ideal to store them securely. We can achieve this in a GitHub repo by using [secrets](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets). So, click on `settings` tab in your repo and add 2 new secrets:
-- `ADMIN_USERNAME` - store the admin user name in this (e.g. user@contoso.onmicrosoft.com)
-- `ADMIN_PASSWORD` - store the password of that user in this.
+
+Since this action requires sensitive information such as user name, password and encoded certificate for example, it would be ideal to store them securely. We can achieve this in a GitHub repo by using [secrets](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets). So, click on `settings` tab in your repo and add:
+
+- 2 new secrets if `authType` is `password`:
+  - `ADMIN_USERNAME` - store the admin user name in this (e.g. user@contoso.onmicrosoft.com)
+  - `ADMIN_PASSWORD` - store the password of that user in this.
+
+- 3 new secrets if `authType` is `certificate`:
+  - `AAD_APP_ID` - store the Azure AD application ID used for authentication
+  - `CERTIFICATE_ENCODED` - store the Base64-encoded string of the certificate stored in the Azure AD application
+  - `CERTIFICATE_PASSWORD` - store the certificate password
```

Comment:
As `APP_ID` can also be used with password login, let's not suggest that it is just specific to certificate authentication.

Let's create a we have a section that calls out `APP_ID` and `TENANT` separately.

```suggestion
- 2 new secrets if `authType` is `certificate`:
  - `CERTIFICATE_ENCODED` - store the Base64-encoded string of the certificate stored in the Azure AD application
  - `CERTIFICATE_PASSWORD` - store the certificate password

- 2 new secrets if using a custom Azure AD identity:
  - `APP_ID` - store App ID of the Azure AD application to use for authentication
  - `TENANT` - store the ID of the tenant from which accounts should be able to authenticate
 
```